pkg/bundle/bundle.go

```diff
@@ -199,7 +199,11 @@ func (b *bundle) reconcileStatusPatch(ctx context.Context, req ctrl.Request) (re
 				Kind:       "ConfigMap",
 			},
 		}
-		err := b.directClient.List(ctx, configMapList, &client.ListOptions{})
+		err := b.directClient.List(ctx, configMapList, &client.ListOptions{
```

Comment:
Why do we need to list directly from kube apiserver? We know which configmaps we're interested in in each namespace as the configmap name is static. I think we should get them by name from cache. Else if there is a reason for wanting to list all that are labelled, then probably better to cache all the labelled ones in full (rather than just partial metadata).

Reply:
I reworked the caching logic in the latest commit.